### PR TITLE
[Fix-10392][SqlTask]: Fix list param error when use sql task

### LINKED ISSUE #10392: [Bug] [SQL] SQL Task Node Custom Parameters run failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

when i using Custom Parameters of SQL Task Node with OUT type, sql query result is more than one row, in downstream sql using the parameters failed, if upstream sql result is one row, it work fine.
failed work flow:
image

work flow success
image

### What you expected to happen

SQL Task Custom parameters work fine

### How to reproduce

add two sql task node, in first node query select and add custom parameter OUT type, in donwstream sql using the parameters 

### Anything else

_No response_

### Version

3.0.0-beta-1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
## Purpose of the pull request

fix list param error when use sql task. close https://github.com/apache/dolphinscheduler/issues/10392

## Brief change log
add replaceListParameter to replace sqlBind.

### CODE REVIEW

**Comment by stalary** [comment]
PTAL @zhongjiajie @caishunfeng

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/ParameterUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+public class ParameterUtilsTest {
+
+    @Test
+    public void replaceListParameter() {
+        Map<Integer, Property> params = new HashMap<>();
+        params.put(1, new Property(null, null, DataType.LIST, JSONUtils.toJsonString(Lists.newArrayList("c1", "c2", "c3"))));
+        params.put(2, new Property(null, null, DataType.DATE, "2020-06-30"));
+        params.put(3, new Property(null, null, DataType.LIST, JSONUtils.toJsonString(Lists.newArrayList("d1", "d2", "d3"))));
+        String sql = ParameterUtils.replaceListParameter(params, "select * from test where col1 in (?) and date=? and col2 in (?)");
```
Comment (zhuangchong) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/ParameterUtilsTest.java]: I suggest that the sql statement is written in accordance with the normal standard,
```
select * from test where col1 in (?,?,?) and date=? and col2 in (?,?,?)
```

Comment (stalary) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/ParameterUtilsTest.java]: This is a little more complicated and requires changes to both sqlBinds and setInParameter, I'll change it later if necessary

Comment (zhuangchong) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/ParameterUtilsTest.java]: The logic of this step can be placed in the `setSqlParamsMap` method, and the logic replaced by `?` is also put in.

https://github.com/apache/dolphinscheduler/blob/a20f31e6db6f12eb08cd45e8f59b392407346a01/dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java#L428-L433

Comment (stalary) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/parser/ParameterUtilsTest.java]: Ok, thanks for the help, I will modify it later.

**Comment by stalary** [comment]
@zhuangchong Please review it again.

**Review by zhuangchong — APPROVED** [review]
LGTM.

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**